pkg/loki/config_wrapper.go

```diff
@@ -96,85 +99,141 @@ func (c *ConfigWrapper) ApplyDynamicConfig() cfg.Source {
-	r.Ruler.Ring.InstanceID = lc.ID
-	r.Ruler.Ring.InstanceInterfaceNames = lc.InfNames
-	r.Ruler.Ring.NumTokens = lc.NumTokens
-	r.Ruler.Ring.KVStore.Store = s
-	r.Ruler.Ring.KVStore.StoreConfig = sc
+	r.Ruler.Ring.HeartbeatPeriod = rc.HeartbeatPeriod
+	r.Ruler.Ring.InstancePort = rc.InstancePort
+	r.Ruler.Ring.InstanceAddr = rc.InstanceAddr
+	r.Ruler.Ring.InstanceID = rc.InstanceID
+	r.Ruler.Ring.InstanceInterfaceNames = rc.InstanceInterfaceNames
+	r.Ruler.Ring.NumTokens = defaultNumTokens
+	r.Ruler.Ring.KVStore.Store = rc.KVStore.Store
+	r.Ruler.Ring.KVStore.StoreConfig = rc.KVStore.StoreConfig
 
 	// Query Scheduler
 	r.QueryScheduler.SchedulerRing.HeartbeatTimeout = rc.HeartbeatTimeout
-	r.QueryScheduler.SchedulerRing.HeartbeatPeriod = lc.HeartbeatPeriod
-	r.QueryScheduler.SchedulerRing.InstancePort = lc.Port
-	r.QueryScheduler.SchedulerRing.InstanceAddr = lc.Addr
-	r.QueryScheduler.SchedulerRing.InstanceID = lc.ID
-	r.QueryScheduler.SchedulerRing.InstanceInterfaceNames = lc.InfNames
-	r.QueryScheduler.SchedulerRing.InstanceZone = lc.Zone
+	r.QueryScheduler.SchedulerRing.HeartbeatPeriod = rc.HeartbeatPeriod
+	r.QueryScheduler.SchedulerRing.InstancePort = rc.InstancePort
+	r.QueryScheduler.SchedulerRing.InstanceAddr = rc.InstanceAddr
+	r.QueryScheduler.SchedulerRing.InstanceID = rc.InstanceID
+	r.QueryScheduler.SchedulerRing.InstanceInterfaceNames = rc.InstanceInterfaceNames
+	r.QueryScheduler.SchedulerRing.InstanceZone = rc.InstanceZone
 	r.QueryScheduler.SchedulerRing.ZoneAwarenessEnabled = rc.ZoneAwarenessEnabled
-	r.QueryScheduler.SchedulerRing.KVStore.Store = s
-	r.QueryScheduler.SchedulerRing.KVStore.StoreConfig = sc
-	f, err = tokensFile(r, "scheduler.tokens")
+	r.QueryScheduler.SchedulerRing.KVStore.Store = rc.KVStore.Store
+	r.QueryScheduler.SchedulerRing.KVStore.StoreConfig = rc.KVStore.StoreConfig
+
+	// Compactor
+	r.CompactorConfig.CompactorRing.HeartbeatTimeout = rc.HeartbeatTimeout
+	r.CompactorConfig.CompactorRing.HeartbeatPeriod = rc.HeartbeatPeriod
+	r.CompactorConfig.CompactorRing.InstancePort = rc.InstancePort
+	r.CompactorConfig.CompactorRing.InstanceAddr = rc.InstanceAddr
+	r.CompactorConfig.CompactorRing.InstanceID = rc.InstanceID
+	r.CompactorConfig.CompactorRing.InstanceInterfaceNames = rc.InstanceInterfaceNames
+	r.CompactorConfig.CompactorRing.InstanceZone = rc.InstanceZone
+	r.CompactorConfig.CompactorRing.ZoneAwarenessEnabled = rc.ZoneAwarenessEnabled
+	r.CompactorConfig.CompactorRing.KVStore.Store = rc.KVStore.Store
+	r.CompactorConfig.CompactorRing.KVStore.StoreConfig = rc.KVStore.StoreConfig
+}
+
+// useIngesterRingConfig will override all other rings to use the ingester's ring configuration
+func useIngesterRingConfig(r, defaults *ConfigWrapper) {
+	appendLoopbackInterface(r, defaults)
+
+	ingesterRingCfg := util.CortexLifecyclerConfigToRingConfig(r.Ingester.LifecyclerConfig)
+	numTokens := r.Ingester.LifecyclerConfig.NumTokens
+
+	// we're using the ingester config here, so never need to override it
```

Comment:
Hey, I was thinking since you are passing the `ingester`, wouldn't it be fine to allow overriding the ingester? In such a scenario you'd be overriding the ingester with the ingester itself, so looks fine.

Reply:
yeah, I like that idea, I'm going to look into removing the whole third param